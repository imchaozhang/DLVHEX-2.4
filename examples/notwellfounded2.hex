dom(x).
dom(xx).

% disjunction a v b v c v d.
foo(xx) v bar(xx) v baz(xx) v d.

% a :- b
bar(x) :- bar(xx).
foo(X) :- &testConcat[Y,x](X), bar(Y), dom(X). 
% b :- a
foo(x) :- foo(xx).
bar(X) :- &testConcat[Y,x](X), foo(Y), dom(X). 
% a :- c
baz(x) :- baz(xx).
foo(X) :- &testConcat[Y,x](X), baz(Y), dom(X). 

% compare with the following program:
% (this program requires sigma-p-2, so it cannot be processed with clingo)
% a v b v c v d.
% a :- b.
% b :- a.
% a :- c.

% the result should be: {a,b} {d}
% hence (translated to foo(xx), bar(xx), baz(xx), and d) it should be
% {foo(xx),bar(xx)} and {d}
