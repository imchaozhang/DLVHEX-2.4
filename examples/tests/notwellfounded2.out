{foo(xx), bar(xx), foo(x), bar(x)}
{d}
